enhancements/single-node/aos-sno-pair.md

```diff
@@ -0,0 +1,143 @@
+- One or more identical Kubernetes Deployments are created on both OCP clusters, and a Pacemaker [Resource Agent](https://github.com/beekhof/pcmk/blob/master/k8sDeployment) is used to manage the replica count of these Deployments in order to ensure the singleton workload instance is always running on a healthy peer.
+- Each Kubernetes Deployment can be configured to prefer a particular OCP cluster in order to make effective use of the hardware during normal operation.
+
+
+### User Stories
+TBD
+
+### Implementation Details/Notes/Constraints
+
+#### Normal Operation
+Two nodes that communicate with each other and co-ordinate such that k8s Deployments are scaled up on at most one peer.
+The RHEL-HA cluster performs periodic health checking of Deployments at two levels, frequently checking the number of requested replicas is correct, but also checking the number of active replicas at a lower frequency in order to capture additional soft-failure scenarios where the scheduler is unable to satisfy the replica count.
+
+It is imagined that most Deployments would be scaled between 0 and 1 but the minimum and maximum counts are admin configurable for added flexibility.
+
+Rather than creating a one-size-fits-all 2 node architecture, supportable customer deployments must include a minimum number of [redundancy options](https://drive.google.com/file/d/1aw_78490rTQGsxEFcZytMe8lasZErLMm/view?usp=sharing) based on the constraints of their environment.
+
+#### Network Redundancy
+Bonding or Knet can be configured with redundant network paths in environments where this is possible.  This helps ensure that the RHEL-HA cluster does not attempt recovery until all communication paths between the peers have been exhausted.
+
+Care must be taken to ensure that paths are truly independent and avoid single points of failure (SPoF). Crossover cables are often the simplest way of meeting this requirement and are therefore highly encouraged.
+
+#### Fencing
+In the event of a peer failure, any **workloads requiring singleton semantics** (such as StatefulSets or kubevirt VMs) require the existing node to be put into a safe state (stopped or otherwise known to be not running the workloads) before they can be started on the surviving peer.  The traditional term for this is “**fencing**” and it is an **optional** part of RHEL-HA clusters.
+
+Customer environments without any singleton workloads may not require fencing, but need to be comfortable with the possibility of workloads being active on both peers at the same time.
+
+For customers that do have singleton workloads, the RHEL-HA stack supports a wide variety of BMCs including those conforming to the IPMI and RedFish standards, as well as SSH and libvirt for testing.
```

Comment:
What API will be provided for configuring access to those BMCs?

Reply:
RHEL-HA includes a python based `fence-agents` package that the RHEL cluster uses to interface with BMC hardware.  For configuring the RHEL cluster, there is a `pcs` command line tool that we will need to drive or wrap. possibly based on the contents of a new CRD that describes the rest of the RHEL cluster (including peer ips).

Reply:
Is there going to be a separate enhancement to describe that API, or should it be included here?

Reply:
The CRD will be included here once we finalize the details